astropy/utils/mixins.py

```diff
@@ -0,0 +1,71 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""A module containing miscellaneous mixin classes."""
+
+from typing import Mapping, Optional
+
+
+__all__ = ["DisplayFormatAttributeMixin"]
+
+
+class DisplayFormatAttributeMixin:
+    """Mixin class for storing format-specific representations of an object.
+
+    Attributes
+    ----------
+    _format : dict[str, str]
+        A mapping to format-specific representations of this object.
+        For example, for the ``Ohm`` unit in :mod:`astropy.units`, it might be
+        nice to have it displayed as ``\\Omega`` by the ``latex`` formatter.
+        In that case, `format` argument should be set to::
+
+            {'latex': r'\\Omega'}
+    name : str
+        The fallback value if a format-specific representation does not exist
+        in the `format` registry.
+
+    Methods
+    -------
+    get_format_name(format='latex')
+        Get a format-specific representation for this object.
+
+    Examples
+    --------
+    ``DisplayFormatAttributeMixin`` is a mixin class, so it must be subclassed.
+
+        >>> class Example(DisplayFormatAttributeMixin):
+        ...     def __init__(self, name, format=None):
+        ...         self.name = name
+        ...         self._format = format or {}  # always a dict
+
+        >>> ex = Example("example", format=dict(latex=r"ex$^{ample}$"))
+
+    Now we can get the format by name.
+
+        >>> ex.get_format_name("latex")
+        'ex$^{ample}$'
+
+    For non-defined formats, the default is the name of the instance.
+
+        >>> ex.get_format_name("not defined")
+        'example'
+    """
+
+    name: str
+    _format: Mapping[str, str]
+
+    def get_format_name(self, format: str) -> str:
```

Comment:
Now that we're factoring out, I wonder whether this should be changed to `get_name`? Though of course that would imply a deprecation in units, which perhaps is not worth it. On yet the other hand, I'd hope that people use `.to_string()` anyway...

Which makes me wonder slightly: is this really a useful public method? (I realize I should look at the follow-up PRs, but then I'd still have nothing).

Reply:
And possibly to offer a default ``__format__`` that can use the ``_format``.